src/Microsoft.CodeQuality.Analyzers/Core/ApiDesignGuidelines/AvoidEmptyInterfaces.cs

```diff
@@ -44,7 +45,11 @@ public override void Initialize(AnalysisContext analysisContext)
         private static void AnalyzeInterface(SymbolAnalysisContext context)
         {
             var symbol = (INamedTypeSymbol)context.Symbol;
-            if (symbol.TypeKind == TypeKind.Interface && !symbol.GetMembers().Any() && !symbol.AllInterfaces.SelectMany(s => s.GetMembers()).Any())
+
+            if (symbol.TypeKind == TypeKind.Interface &&
+                symbol.GetResultantVisibility() == SymbolVisibility.Public &&
+                !symbol.GetMembers().Any() &&
```

Comment:
Not relevant to this PR, but we should really just use Count > 0 here, this is an ImmutableArray. No need for linq allocations.

Reply:
Done